### ISSUE TITLE
parameter owner in sympa config file not considered by sympa 6.2.38

### ISSUE THREAD

**Issue by bv-unibs** [sympa-community/sympa#530]
Hi, I'm trying to migrate my old mailing list server sympa (6.1.17) to the newer version 6.2.38.

I've imported into the new sympa server the list-data directory with all my mailing lists conf files, and almost everything seems ok, except for something about the owner parameter that seems not to be considered by sympa.

In sympa.log i keep receiving no_owner error for all my lists, but in each config file I have two owner entries like this 

owner
reception mail
profile privileged
email <email_address>
visibility noconceal
gecos <user_name>

In the previous version the data in config files were correctly read, but in this version seems ignored.

I tried to enter the owner by using web interface, but I would like to bypass the web interface, because i need to input these informations for more than 200 mailing lists.

I suppose sympa saves these info into the back-end db, ignoring the config file setup..

Is it possible to instruct sympa to consider the parameter owner in the config file of the lists?

Thanks for your attention

Regards

**Comment by dverdin** [comment]
Hi,
Starting 6.2.34, editors and owners are stored in database only.
If you run the following command, these data should be moved from files to database:
```
sympa.pl --upgrade
```
Cheers!

**Comment by bv-unibs** [comment]
Hi,
I've already tried this command, but I can't use it because I didn't upgrade the old sympa server in place, but I've installed a new server with 6.2.38 version of sympa from scratch, and imported into the new server the old configuration files.

So the command gives me the following output:
**### 
notice main:: Sympa 6.2.38 Started
notice main:: Upgrade process...
notice main:: Current version: 6.2.38; no upgrade is required**

How can I correct the old list configurations for working with new sympa in this scenario? 

regards

**Comment by dverdin** [comment]
Oh, then it's easy: you fake upgrade:

```
sympa.pl --upgrade --from=6.1.17 --to=6.2.38
```
Et voilà ! ;)

**Comment by bv-unibs** [comment]
It worked like a charm, the lists are now updated with the missing parameters..

thank you

**Comment by fpoulain** [comment]
Hi,

Now we have those data in db but still have owner sections in config files.
This may be confusing for admins. Do we have to manually clean the configs ? Is there other list configs param which has been moved to db ?
Didn't found mention to this in https://sympa-community.github.io/manual/upgrade/notes.html

Best regards.
François

**Comment by ikedas** [comment]
> Hi,
> 
> Now we have those data in db but still have owner sections in config files.
> This may be confusing for admins. Do we have to manually clean the configs ? Is there other list configs param which has been moved to db ?
> Didn't found mention to this in https://sympa-community.github.io/manual/upgrade/notes.html

Config parameters `owner` and `editor` were obsoleted as of the version dverdin wrote.  As these parameters in `config` file are ignored, you need not remove them from `config` by yourself: They may be removed by the future update.

However note that these parameters are available in `config.tt2` of list creation templates: When a list is created, `owner` and `editor` in config generated from `config.tt2` file are imported into database.

**Comment by iulistadmin** [comment]
Are there instances in which Sympa reloads the owner and/or editor information from the configs? We upgraded from 6.1.11 to 6.2.60 and we see that some list's configs still have the owner and editor information in them. Wasn't this information supposed to have been removed during the upgrade? 

The correct owner information is in the Sympa database, but in some cases the list's configs are inconsistent with the database. So, in the configs there may be three owners, but the database may include four, with the database being accurate. We have had some lists occasionally see the fourth owner removed, and we suspect that some Sympa process or the --upgrade command we have been running as we've migrated lists to the new environment has reloaded the config files from the old lists, removing the fourth owner from the database. In some cases, it appears owners, editors, etc. may be getting ingested from dump files in the lists's directory, so have been removing them to prevent unwanted changes.

**Comment by ikedas** [comment]
Hi @iulistadmin , 
After when Sympa has been upgraded to recent  version, it no longer will use `owner` and/or `editor` paragraphs included in list `config` file: They are simply ignored.

> We have had some lists occasionally see the fourth owner removed, and we suspect that some Sympa process or the --upgrade command we have been running as we've migrated lists to the new environment has reloaded the config files from the old lists, removing the fourth owner from the database. In some cases, it appears owners, editors, etc. may be getting ingested from dump files in the lists's directory, so have been removing them to prevent unwanted changes.

I'd like to clarify: Have you removed whether `owner`/`editor` paragraphs in `config` file or `owner.dump`/`editor.dump` files?

**Comment by iulistadmin** [comment]
With these lists, we have noticed that the config files still contain the owner/editor information in them. Some lists have had owner.dump/editor.dump files in their directories, as well. So, we suspected that these config files may have been re-read and overwrote the information in the database with the old information from the config (i.e., list config or the dump files). We even removed all dump files recently to avoid this issue, but we have not directly modified the list configs with owner/editor data therein.
